### ISSUE TITLE
VM does not exit after upgrade from 3.7.0 to 3.8.0

### ISSUE THREAD

**Issue by McBluna** [w3stling/rssreader#170]
I've reduced the code to a minimum. Exit is printed to console, but VM keeps running.
Environment is Java 17.0.11 running on Windows 11.

```
	public static void main(String[] args) throws IOException {
		RssReader rssReader = new RssReader();
		List<Item> items = rssReader.read("https://www.tagesschau.de/infoservices/alle-meldungen-100~rss2.xml").toList();
		System.out.println(items);
		System.out.println("exit");
	}
```
![grafik](https://github.com/user-attachments/assets/58440164-7fa0-48f5-9d52-10ea11516cbc)

**Comment by McBluna** [comment]
Btw changing to Java 21.0.3 didn't change anything.

**Comment by McBluna** [comment]
Another try but still no working
```
	public static void main(String[] args) throws IOException {
		RssReader reader = new RssReader();
		try (Stream<Item> rssFeed = reader.read(URL)) {
		}
		System.out.println("Exit");
	}
```

**Comment by w3stling** [comment]
Fixed by PR #171
Please try release [v3.8.1](https://github.com/w3stling/rssreader/releases/tag/v3.8.1)

**Comment by McBluna** [comment]
Sorry I wasn't at home. It's working again with 3.8.1.
Thank you so much.
